autopush/bridge/bridge.py

```diff
@@ -0,0 +1,67 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from apns_ping import APNSBridge
+from gcm_ping import GCMBridge
+
+from twisted.python import log
+
+__all__ = ["BridgeUndefEx", "BridgeFailEx", "Bridge"]
+
+
+class BridgeUndefEx(Exception):
+    pass
+
+
+class BridgeFailEx(Exception):
+    pass
+
+
+class Bridge(object):
+    storage = None
+
+    def __init__(self, storage, settings):
+        if storage is None:
+            raise BridgeUndefEx("No storage defined for Bridge")
+        self.storage = storage
+        self.gcm = None
+        self.apns = None
+        if settings.get('gcm'):
+            self.gcm = GCMBridge(settings.get('gcm'), storage)
+        if settings.get('apns'):
+            self.apns = APNSBridge(settings.get('apns'), storage)
+
+    def register(self, uaid, connect):
+        # Store the connect string to the database
+        if self.storage is None:
+            raise BridgeUndefEx("No storage defined for Bridge")
+        if connect is None or connect is False:
+            return None
+        try:
+            if self.storage.register_connect(uaid, connect) is False:
+                log.err("Failed to register connection for %s:%s" %
+                        (uaid, connect))
+                raise BridgeFailEx("Unable to register connection")
+        except Exception, e:
+            log.err(e)
+            raise
+        return True
+
+    def ping(self, uaid, version, data, connectInfo):
+        if connectInfo is None or connectInfo is False:
```

Comment:
More trapping if other falsy values happen to show up, but collapsing to if not connectInfo should suffice.